src/Evolution/DgSubcell/TwoMeshRdmpTci.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+
+#include "DataStructures/Variables.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace evolution::dg::subcell {
+/*!
+ * \brief Troubled cell indicator using a relaxed discrete maximum principle,
+ * comparing the solution on two grids at the same point in time.
+ *
+ * Checks that the subcell solution \f$\underline{u}\f$ and the DG solution
+ * \f$u\f$ satisfy
+ *
+ * \f{align*}{
+ * \min(u)-\delta \le \underline{u} \le \max(u)+\delta
+ * \f}
+ *
+ * where
+ *
+ * \f{align*}{
+ * \delta = \max\left[\delta_0, \epsilon(\max(u) - \min(u))\right]
+ * \f}
+ */
+template <typename... EvolvedVarsTags>
+bool two_mesh_rdmp_tci(
+    const Variables<tmpl::list<EvolvedVarsTags...>>& dg_evolved_vars,
+    const Variables<tmpl::list<Tags::Inactive<EvolvedVarsTags>...>>&
+        subcell_evolved_vars,
+    const double rdmp_delta0, const double rdmp_epsilon) noexcept {
+  bool cell_is_troubled = false;
+  tmpl::for_each<tmpl::list<EvolvedVarsTags...>>(
+      [&cell_is_troubled, &dg_evolved_vars, rdmp_delta0, rdmp_epsilon,
+       &subcell_evolved_vars](auto tag_v) noexcept {
+        if (cell_is_troubled) {
+          return;
+        }
+        using std::max;
+        using std::min;
+
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        using inactive_tag = Tags::Inactive<tag>;
+        const auto& dg_var = get<tag>(dg_evolved_vars);
+        const auto& subcell_var = get<inactive_tag>(subcell_evolved_vars);
+
+        for (auto dg_it = dg_var.begin(), subcell_it = subcell_var.begin();
+             dg_it != dg_var.end() and subcell_it != subcell_var.end();
+             (void)++dg_it, (void)++subcell_it) {
+          ASSERT(not cell_is_troubled,
+                 "If a cell has already been marked as troubled during the "
+                 "single-level RDMP TCI, we should not be continuing to check "
```

Comment:
Should "single level" -> "two mesh"? I don't think you've used "single level" elsewhere in this code...